### ISSUE TITLE
TIDAL Connect does not work with current adapter version

### ISSUE THREAD

**Issue by mllrmnn** [withstu/ioBroker.heos#331]
Hi,

since the Firmware from 24.07.2024 - U34 ([New HEOS App and firmware update available](https://support-uk.marantz.com/app/answers/detail/a_id/7214/~/new-heos-app-and-firmware-update-available)) the TIDAL Connect feature ist available for selected AVR Receivers, via the HEOS Firmware.

I disovered, that if I play something over TIDAL Connect, the HEOS adapter does only update/refresh the following objects:
```
- heos.0.players.XXXXX.current_elapsed_s
- heos.0.players.XXXXX.current_elapsed
- heos.0.players.XXXXX.current_duration_s
- heos.0.players.XXXXX.current_duration
```
all other `current_*` objects does NOT get updated/refreshed. All other Sources work as they should.

Is it possible to implement the TIDAL Connect Source "as a valid Source" in the adapter to obtain all the other states as well?

If you need any other informations, let me know!

HEOS Adapter Version: 2.2.1
Plattform: docker (official image - v9.1.2)
Betriebssystem: linux
Architektur: x64
CPUs: 8
Geschwindigkeit: 2100 MHz
Modell: Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
RAM: 15.3 GB
Node.js: v18.20.2
NPM: 10.7.0

**Comment by withstu** [comment]
Thank you for raising this issue. I've quickly looked into it and it seems that the HEOS API has a bug here. When a song is played with Tidal Connect the API is sending the Source ID (SID) 0, which is not a valid ID when you look into their API documentation. The correct ID should be 10: https://rn.dmglobal.com/usmodel/HEOS_CLI_ProtocolSpecification-Version-1.17.pdf. As I'm not expecting that this is fixed by Denon soon, I'm trying to implement a workaround. Stay tuned.

**Comment by withstu** [comment]
fixed in 2.2.2

**Comment by mllrmnn** [comment]
Thanks, it works!
